### DISCUSSION TITLE
How to use absolute URLs for all resource on a static page

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#47454] [Help]
### Summary

For the integration of en external service the provision of a static page is required. All the resources of that document should be provided using URLs instead of absolute paths. The following markup doesn't work since the styles aren't being loaded. 

```html
<!DOCTYPE html>
<html lang="en">
<head>
  <!--###start_head###-->
  <meta charSet="utf-8" />
  <meta name="viewport" content="width=device-width" />
  <title>My Page</title>
  <link rel="icon" href="/assets/icon/favicon.png">
  <link rel="preload" href="/_next/static/css/{hash}.css" as="style" />
  <link rel="stylesheet" href="/_next/static/css/{hash}.css" data-n-g="" />
  <script defer="" nomodule="" src="/_next/static/chunks/polyfills-{hash}.js"></script>
  <script src="/_next/static/chunks/webpack-{hash}.js" defer=""></script>
  <script src="/_next/static/chunks/framework-{hash}.js" defer=""></script>
  <script src="/_next/static/chunks/main-{hash}.js" defer=""></script>
  <script src="/_next/static/chunks/pages/_app-{hash}.js" defer=""></script>
  <script src="/_next/static/chunks/123-{hash}.js" defer=""></script>
  <script src="/_next/static/chunks/456-{hash}.js" defer=""></script>
  <!--###end_head###-->
</head>
<body>
  <!--###start_body###-->
  <div>stuff</div>
  <!--###end_body###-->
</body>
</html>
```

While this would be possible for the manually inserted resources such as favicons, this is not possible for the resources starting with `_next/...`. Lets leave the fact aside, that this would apply to all pages.

The external services runs on another host and uses this static page as a skeleton, where only the frame is used, thus the placeholders `###start_head###` as html comments.

**In a nutshell:**
How to provide URLs instead of absolute paths to the resources of a static page / certain route e.g.: https://mysite.com/static/service.html

**Comment by philwolstenholme** [comment]
I'm not sure there's a way to do this just with Next configuration.

But…

If you can get Next to build a static version of the page then you could do this using [PostHTML](https://github.com/posthtml/posthtml) in a second build step, I imagine.

Or something like Cloudflare workers or a Netlify function, maybe a Vercel edge function could rewrite the HTML too. You'd need a HTML rewriter like https://github.com/worker-tools/parsed-html-rewriter (JS) or https://deno.land/x/html_rewriter (Deno).
